From alan@linux.intel.com  Mon Dec  7 16:58:09 2009
From: Alan Cox <alan@linux.intel.com>
Date: Mon, 30 Nov 2009 13:16:30 +0000
Subject: tty: sdio_uart: Fix the locking on "func" for new code
To: greg@kroah.com
Message-ID: <20091130131630.26701.97011.stgit@localhost.localdomain>


The new dtr_rts function didn't take the port->func lock as it should
so add use of the lock there.


Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/mmc/card/sdio_uart.c |    4 ++++
 1 file changed, 4 insertions(+)


--- a/drivers/mmc/card/sdio_uart.c
+++ b/drivers/mmc/card/sdio_uart.c
@@ -595,10 +595,14 @@ static void uart_dtr_rts(struct tty_port
 {
 	struct sdio_uart_port *port =
 			container_of(tport, struct sdio_uart_port, port);
+	int ret = sdio_uart_claim_func(port);
+	if (ret)
+		return;
 	if (onoff == 0)
 		sdio_uart_clear_mctrl(port, TIOCM_DTR | TIOCM_RTS);
 	else
 		sdio_uart_set_mctrl(port, TIOCM_DTR | TIOCM_RTS);
+	sdio_uart_release_func(port);
 }
 
 /**
